### PR TITLE
Removed Unused Code & Updating Form ?

### PR DESCRIPTION
**Pull request by HeIp**

### CODE REVIEW

File: src/SignUp.js [hunk 1]
```diff
@@ -11,23 +10,34 @@ export default function SignUp() {
...
+          ref={register({ required: true, minLength: 6 })}
+      />
+      {errors.userName && <p>Username field is required & Length needs to be greater than 5</p>}
+
       <label className="signup-label">Password</label>
       <input
         className="signup-input"
-        name="exampleRequired"
-        ref={register({ required: true, maxLength: 10 })}
+        name="passWord"
```
Comment (FultonG) [inline src/SignUp.js]: password is one word. No need to camelCase

File: src/Components/MainPage/MainPage.js [hunk 2]
```diff
@@ -28,6 +33,9 @@ const MainPage = () => {
     fetchData();
   }, []);
 
+  /*
+  Runs on load and if current token changes
+   */
```
Comment (FultonG) [inline src/Components/MainPage/MainPage.js]: The comment is not really telling me what it does just when it runs. Not super useful

File: src/Components/MainPage/MainPage.js [hunk 3]
```diff
@@ -19,6 +19,11 @@ const MainPage = () => {
     }
   }, [currentToken])
 
+  /*
+  Onload get tokens,
+  save tokens,
+  set current token to first token
+  */
```
Comment (FultonG) [inline src/Components/MainPage/MainPage.js]: This is better

File: src/SignUp.js [hunk 4]
```diff
@@ -11,23 +10,34 @@ export default function SignUp() {
...
+      {errors.userName && <p>Username field is required & Length needs to be greater than 5</p>}
+
       <label className="signup-label">Password</label>
       <input
         className="signup-input"
-        name="exampleRequired"
-        ref={register({ required: true, maxLength: 10 })}
+        name="password"
+        ref={register({ required: true, minLength: 8 })}
       />
```
Comment (FultonG) [inline src/SignUp.js]: this input need to be of type password

File: src/formstyles.css [hunk 5]
```diff
@@ -30,7 +30,7 @@
 }
 
 .signup-button[type = "submit"]{
-  background: #fddb92;
+  background: #998DA0;
```
Comment (FultonG) [inline src/formstyles.css]: This color is too dark. it's hard to read the text of the button

File: src/Services/API.js [hunk 6]
```diff
@@ -17,6 +17,16 @@ const API = {
     getStopsByRoute: async function (routeId) {
         let response = await axios.get(`${URL}/stops/find?id=${routeId}`);
         return response.data;
+    },
+    registerUser: async function (username, password, email) {
+        // let response = await axios.post(`${URL}/register?user=${username}&pwd=${password}&email=${email}`);
```
Comment (FultonG) [inline src/Services/API.js]: ?

File: src/SignUp.js [hunk 7]
```diff
@@ -1,33 +1,80 @@
...
+    }
+    /**
+     * Username was not taken,
+     * Gets registered in backend
+     * Redirects to Home page('/')
+     */
+    else
+    {
+      clearError("user");
+      alert("Succesfuly created an account!");
```
Comment (FultonG) [inline src/SignUp.js]: Actually create a modal for this not just an alert

File: src/SignUp.js [hunk 8]
```diff
@@ -1,33 +1,80 @@
...
+
   const onSubmit = data => {
-    console.log(data);
-  }; 
+    const {pwd, email, user} = data;
+    validate(data);
+  };
 
-  console.log(watch("example")); 
+  //TODO figure out redirrect
```
Comment (FultonG) [inline src/SignUp.js]: is redirect not working?

Comment (HeIp) [reply src/SignUp.js]: It was just removed the todo

File: src/Services/API.js (hunk 6 above)
Comment (HeIp) [reply src/Services/API.js]: Removed comment

File: src/Services/API.js [hunk 9]
```diff
@@ -17,6 +17,15 @@ const API = {
     getStopsByRoute: async function (routeId) {
         let response = await axios.get(`${URL}/stops/find?id=${routeId}`);
         return response.data;
+    },
+    registerUser: async function (username, password, email) {
+        let response = await axios.post(`${URL}/register?user=${username}&pwd=${password}&email=${email}`)
+            .catch(function (error)
+            {
+            console.log("This is the error log:", error)
```
Comment (FultonG) [inline src/Services/API.js]: stop leaving `console.log`

Comment (HeIp) [reply src/Services/API.js]: Done Senpai

**Pull request merged**